### ISSUE TITLE
Support CoreDNS version 1.8.0+

### ISSUE THREAD

**Issue by colearendt** [traefik/mesh#787]
## Feature Request

Support CoreDNS version 1.8.0+

### Proposal

The current implementation seems to fail on CoreDNS version 1.8.0. I propose supporting that version 😄 

https://github.com/traefik/mesh/blob/e53767daee443f55338a4ffcf07e66b39afb630f/pkg/dns/client.go#L40

https://github.com/traefik/mesh/blob/e53767daee443f55338a4ffcf07e66b39afb630f/pkg/dns/client.go#L100-L104

### Background

Installing traefik-mesh into a cluster with CoreDNS 1.8.0 fails with:
```
$ kubectl logs traefik-mesh-controller-6c4768b4c7-dm2w5 -c traefik-mesh-prepare
2021/01/28 06:10:04 command prepare error: unable to find suitable DNS provider: unsupported CoreDNS version "1.8.0"
```

Interestingly, `traefik-mesh-cleanup` _also_ cannot complete properly due to the DNS version, which makes my helm deployment failure clean up improperly.

```
# kubectl logs traefik-mesh-cleanup-k6ptr
2021/01/28 06:00:33 command cleanup error: error encountered during DNS restore: unsupported CoreDNS version "1.8.0"
```

### Workarounds

Downgrade CoreDNS

**Comment by 0rax** [comment]
This is definitely an issue preventing us from upgrading our cluster to Kubernetes 1.20 on EKS as the default CoreDNS version there is `1.8.0`.

I will check and see if there is any major change preventing mesh from working with CoreDNS 1.8 by default just in case we just need to update the code to support it.

Was there any research already done on the subject by someone at Traefik ?

**Comment by traefiker** [comment]
Closed by #790.
